scripts/gen_regex.py

```diff
@@ -4,89 +4,74 @@
 from pkg_resources import resource_filename
 
 
+CATEGORIES = [unicodedata.category(chr(i)) for i in range(0x110000)]
 DATA_PATH = pathlib.Path(resource_filename('wordfreq', 'data'))
 
 
+def func_to_regex(func):
+    """
+    Given a function that returns True or False for a numerical codepoint,
+    return a regex character class accepting the characters resulting in True.
+    Ranges separated only by unassigned characters are merged for efficiency.
```

Comment:
If by "an unassigned codepoint" you mean "an un-accepted, unassigned codepoint", then I don't think so.  Note that when *building* the regex it uses `r[1]`, which is only updated for accepted codepoints.